src/stan/variational/advi.hpp

```diff
@@ -334,6 +357,14 @@ namespace stan {
 
         // draw more samples from posterior and write on subsequent lines
         if (out_stream_) {
+          if (print_stream_) {
```

Comment:
if `out_stream_` doesn't exist, then the algorithm doesn't need to draw any samples. (that's currently how it works)